infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -650,29 +650,20 @@ private void readAddressSpace(SegmentHandle sh) throws IOException {
      * @param address The address of the entry.
      * @return The log unit entry at that address, or NULL if there was no entry.
      */
-    private LogData  readRecord(SegmentHandle sh, long address)
+    private LogData readRecord(SegmentHandle sh, long address)
             throws IOException {
-        FileChannel fc = null;
-        try {
-            fc = getChannel(sh.fileName, true);
-            AddressMetaData metaData = sh.getKnownAddresses().get(address);
-            if (metaData == null) {
-                return null;
-            }
-
-            fc.position(metaData.offset);
+        FileChannel fc = sh.getReadChannel();
+        AddressMetaData metaData = sh.getKnownAddresses().get(address);
+        if (metaData == null) {
+            return null;
+        }
 
-            try {
-                ByteBuffer entryBuf = ByteBuffer.allocate(metaData.length);
-                fc.read(entryBuf);
-                return getLogData(LogEntry.parseFrom(entryBuf.array()));
-            } catch (InvalidProtocolBufferException e) {
-                throw new DataCorruptionException();
-            }
-        } finally {
-            if (fc != null) {
-                fc.close();
-            }
+        try {
+            ByteBuffer entryBuf = ByteBuffer.allocate(metaData.length);
+            fc.read(entryBuf, metaData.offset);
```

Comment:
I cannot tell from the code review if we handle all the runtime exceptions  for fc.read appropriately. We might be but since I could not tell, I wanted to ask.

IllegalArgumentException - If the position is negative
NonReadableChannelException - If this channel was not opened for reading
ClosedChannelException - If this channel is closed
AsynchronousCloseException - If another thread closes this channel while the read operation is in progress
ClosedByInterruptException - If another thread interrupts the current thread while the read operation is in progress, thereby closing the channel and setting the current thread's interrupt status